src/modules/endpoint/sagas/loadCityContent.js

```diff
@@ -11,80 +11,90 @@ import type { FetchCitiesActionType, InitializeCityContentActionType } from '../
 import loadLanguages from './loadLanguages'
 import ResourceURLFinder from '../ResourceURLFinder'
 import buildResourceFilePath from '../buildResourceFilePath'
+import { ContentLoadCriterion } from '../ContentLoadCriterion'
+import DatabaseContext from '../DatabaseContext'
 import AppSettings from '../../settings/AppSettings'
 import NetInfo from '@react-native-community/netinfo'
 
-const MAX_CONTENT_AGE = 24
-
+/**
+ *
+ * @param dataContainer
+ * @param newCity
+ * @param newLanguage
+ * @param criterion
+ * @returns Returns a saga which returns whether all data was loaded and
+ *          the inputs (like newCity, newLanguage) are valid
+ */
 export default function * loadCityContent (
   dataContainer: DataContainer, newCity: string, newLanguage: string,
-  forceUpdate: boolean, shouldRefreshResources: boolean
-): Saga<void> {
-  const appSettings = new AppSettings()
-  yield call(appSettings.setSelectedCity, newCity)
+  criterion: ContentLoadCriterion
+): Saga<boolean> {
+  if (!criterion.peeking()) {
+    const appSettings = new AppSettings()
+    yield call(appSettings.setSelectedCity, newCity)
+  }
 
-  yield call(dataContainer.setContext, newCity, newLanguage)
+  const context = new DatabaseContext(newCity, newLanguage)
 
   const fetchCities: FetchCitiesActionType = { type: 'FETCH_CITIES' }
   yield put(fetchCities)
 
-  let lastUpdate: moment | null = null
-  if (dataContainer.lastUpdateAvailable()) {
-    lastUpdate = yield call(dataContainer.getLastUpdate)
-  }
+  const lastUpdate: moment | null = yield call(dataContainer.getLastUpdate, context)
 
   console.debug('Last city content update on ',
     lastUpdate ? lastUpdate.toISOString() : 'never')
 
-  // The last update was more than 24h ago or a refresh should be forced
-  const shouldUpdate = forceUpdate || !lastUpdate ||
-    lastUpdate.isBefore(moment.tz('UTC').subtract(MAX_CONTENT_AGE, 'hours'))
+  const shouldUpdate = criterion.shouldUpdate(lastUpdate)
 
   console.debug('City content should be refreshed: ', shouldUpdate)
 
-  yield call(loadLanguages, newCity, dataContainer, shouldUpdate)
-  const languages = yield call(dataContainer.getLanguages)
+  if (criterion.shouldLoadLanguages()) {
+    yield call(loadLanguages, context, dataContainer, shouldUpdate)
+    const languages = yield call(dataContainer.getLanguages, context)
+
+    const initializeCityContent: InitializeCityContentActionType = {
+      type: 'INITIALIZE_CITY_CONTENT',
+      params: {
+        city: newCity,
+        language: newLanguage,
+        languages
+      }
+    }
+
+    yield put(initializeCityContent)
 
-  const initializeCityContent: InitializeCityContentActionType = {
-    type: 'INITIALIZE_CITY_CONTENT',
-    params: {
-      city: newCity,
-      language: newLanguage,
-      languages
+    if (!languages.map(language => language.code).includes(newLanguage)) {
```

Comment:
We should  maybe keep in mind that it will not be checked whether the linked language is available. This should probably be fixed in NATIVE-249

Reply:
Which linked language? `newLanguage` will have the value of the linked language.

Reply:
Yep, but we can't be sure that language is available in the linked city

Reply:
Oh oke now I think I know what you mean. For example if you put XX as language then it will probably fail here.

Yeah should test this maybe. How did you do handle this with the linking within a city?

Reply:
Not at all, there is a task for this. I think https://issues.integreat-app.de/browse/NATIVE-249